tensorflow/core/kernels/segment_reduction_ops_gpu.cu.h

```diff
@@ -347,7 +347,7 @@ __global__ void SegmentReduceVectorKernel(
         Treducevec block_result =
             x_ok && y_ok ? input_vec[input_idx] : Tvec(initial_value);
         // Apply weights if provided.
-        if (weights && y_ok) block_result *= Tvec(weights[y_idx]);
+        if (weights && y_ok) block_result = block_result * Tvec(weights[y_idx]); // MSVC fix
```

Comment:
nit: we need two spaces before the comment.

Can you add a more descriptive comment, that MSVC does not support compound-assignment operators on device?